src/components/ERC20Balance/balanceMethods/getBalances.js

```diff
@@ -0,0 +1,183 @@
+import BigNumber from "bignumber.js";
+import { fetchDataUsingMulticall } from "./multicall";
+import { getPriceInformation } from "./getPriceInformation";
+import { getChainDetails } from "../../../helpers/getChainDetails";
+import { mockBalances } from "./mockData";
+
+const balanceAbi = [
+  {
+    constant: true,
+    inputs: [{ name: "who", type: "address" }],
+    name: "balanceOf",
+    outputs: [{ name: "", type: "uint256" }],
+    payable: false,
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    constant: true,
+    inputs: [{ name: "addr", type: "address" }],
+    name: "getEthBalance",
+    outputs: [{ name: "balance", type: "uint256" }],
+    payable: false,
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+export const isZeroAddress = (address) => {
+  if (address === "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee") {
+    return true;
+  }
+  return false;
+};
+
+export const getBalances = async (walletAddress, tokens) => {
+  if (tokens.length === 0) {
+    return [];
+  }
+  const { chainId } = tokens[0];
+  tokens.forEach((token) => {
+    if (token.chainId !== chainId) {
+      // eslint-disable-next-line no-console
+      console.warn(`Requested tokens have to be on same chain.`);
+      return [];
+    }
+  });
+
+  const multicallAddress = process.env.REACT_APP_MULTICALL_ADDRESS;
```

Comment:
How would we set this env for many different contract addresses? Wouldn't it be better if it was a config?